src/controllers/krate/search.rs

```diff
@@ -407,6 +397,199 @@ impl<'a> FilterParams<'a> {
 
         Ok(query)
     }
+
+    fn seek_after(&self, seek_payload: &seek::SeekPayload) -> BoxedCondition<'a> {
+        use seek::*;
+
+        let crates_aliased = alias!(crates as crates_aliased);
+        let crate_name_by_id = |id: i32| {
+            crates_aliased
+                .find(id)
+                .select(crates_aliased.field(crates::name))
+                .single_value()
+        };
+        let conditions: Vec<BoxedCondition<'_>> = match *seek_payload {
+            SeekPayload::Name(Name(id)) => {
+                // Equivalent of:
+                // `WHERE name > name'`
+                vec![Box::new(crates::name.nullable().gt(crate_name_by_id(id)))]
```

Comment:
why the `nullable()` here?

Reply:
Without `nullable()`, compiler complains about a type mismatched.
```
error[E0271]: type mismatch resolving `<Grouped<Subselect<SelectStatement<FromClause<Alias<crates_aliased>>, SelectClause<AliasedField<crates_aliased, name>>, NoDistinctClause, ..., ..., ...>, ...>> as Expression>::SqlType == Text`
   --> src/controllers/krate/search.rs:415:44
    |
415 |                 vec![Box::new(crates::name.gt(crate_name_by_id(id)))]
    |                                            ^^ expected `Nullable<Text>`, found `Text`
    |
    = note: expected struct `diesel::sql_types::Nullable<diesel::sql_types::Text>`
               found struct `diesel::sql_types::Text`
    = note: required for `Grouped<Subselect<SelectStatement<FromClause<Alias<crates_aliased>>, SelectClause<AliasedField<crates_aliased, name>>, NoDistinctClause, ..., ..., ...>, ...>>` to implement `AsExpression<diesel::sql_types::Text>`
```